pkg/deploy/controller/deployment/controller.go

```diff
@@ -141,14 +136,34 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 			glog.V(4).Infof("Cancelling %d deployer pods for deployment %s", len(deployerPods), deployutil.LabelForDeployment(deployment))
 			zeroDelay := int64(1)
 			for _, deployerPod := range deployerPods {
-				// Set the ActiveDeadlineSeconds on the pod so it's terminated very soon.
-				if deployerPod.Spec.ActiveDeadlineSeconds == nil || *deployerPod.Spec.ActiveDeadlineSeconds != zeroDelay {
-					deployerPod.Spec.ActiveDeadlineSeconds = &zeroDelay
-					if _, err := c.podClient.updatePod(deployerPod.Namespace, &deployerPod); err != nil {
-						c.recorder.Eventf(deployment, "failedCancellation", "Error cancelling deployer pod %s for deployment %s: %v", deployerPod.Name, deployutil.LabelForDeployment(deployment), err)
-						return fmt.Errorf("couldn't cancel deployer pod %s for deployment %s: %v", deployerPod.Name, deployutil.LabelForDeployment(deployment), err)
+				if deployerPod.Status.Phase == kapi.PodPending {
+					// If the pod hasn't yet been scheduled, just delete it.
+					err := c.podClient.deletePod(deployerPod.Namespace, deployerPod.Name)
+					if err != nil {
+						c.recorder.Eventf(deployment, "failedCancellation", "Error deleting deployer pod %s while cancelling deployment %s: %v", deployerPod.Name, deployutil.LabelForDeployment(deployment), err)
+						return fmt.Errorf("couldn't delete deployer pod %s while cancelling deployment %s: %v", deployerPod.Name, deployutil.LabelForDeployment(deployment), err)
+					}
+					// If we deleted the unscheduled primary deployer pod, just
+					// transition to failed immediately, since the DeployerPodController
+					// has no cleanup to do as the deployment never executed.
+					if deployerPod.Name == deployutil.DeployerPodNameForDeployment(deployment.Name) {
```

Comment:
It should be solved in a different way, really.

On Aug 27, 2015, at 3:55 PM, Dan Mace <notifications@github.com> wrote:

In pkg/deploy/controller/deployment/controller.go
<https://github.com/openshift/origin/pull/4364#discussion_r38132882>:

> -				if deployerPod.Spec.ActiveDeadlineSeconds == nil || *deployerPod.Spec.ActiveDeadlineSeconds != zeroDelay {
> -					deployerPod.Spec.ActiveDeadlineSeconds = &zeroDelay
> -					if _, err := c.podClient.updatePod(deployerPod.Namespace, &deployerPod); err != nil {
> -						c.recorder.Eventf(deployment, "failedCancellation", "Error cancelling deployer pod %s for deployment %s: %v", deployerPod.Name, deployutil.LabelForDeployment(deployment), err)
> -						return fmt.Errorf("couldn't cancel deployer pod %s for deployment %s: %v", deployerPod.Name, deployutil.LabelForDeployment(deployment), err)
> +				if deployerPod.Status.Phase == kapi.PodPending {
> +					// If the pod hasn't yet been scheduled, just delete it.
> +					err := c.podClient.deletePod(deployerPod.Namespace, deployerPod.Name)
> +					if err != nil {
> +						c.recorder.Eventf(deployment, "failedCancellation", "Error deleting deployer pod %s while cancelling deployment %s: %v", deployerPod.Name, deployutil.LabelForDeployment(deployment), err)
> +						return fmt.Errorf("couldn't delete deployer pod %s while cancelling deployment %s: %v", deployerPod.Name, deployutil.LabelForDeployment(deployment), err)
> +					}
> +					// If we deleted the unscheduled primary deployer pod, just
> +					// transition to failed immediately, since the DeployerPodController
> +					// has no cleanup to do as the deployment never executed.
> +					if deployerPod.Name == deployutil.DeployerPodNameForDeployment(deployment.Name) {

Yeah, logs must be it. If we delete, you'll have no way to view the
cancelled deployment logs. Are we okay with that? We get plenty of heat
already for the inaccessibility of deployments logs as is.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4364/files#r38132882>.